Document the repository as being in maintenance only (#1617)
elastic/apm-agent-go: README.md

[![Go Report Card](https://goreportcard.com/badge/go.elastic.co/apm/v2)](https://goreportcard.com/report/go.elastic.co/apm/v2)
[![codecov.io](https://codecov.io/github/elastic/apm-agent-go/coverage.svg?branch=main)](https://codecov.io/github/elastic/apm-agent-go?branch=main)

> **NOTE**: This repository is in maintenance mode. Bug fixes will be worked on,

Reviewer: ```suggestion
**NOTE**: This repository is in maintenance mode. Bug fixes will continue to be applied,
```

Reviewer: and then the same update in `introduction`

but no further development will take place. To replace this agent, we recommend
you to [migrate to the OpenTelemetry Go API and
SDK](https://www.elastic.co/blog/elastic-go-apm-agent-to-opentelemetry-go-sdk),
which provides similar features. In order to help you do a seamless migration,
we recommend using our [OpenTelemetry
Bridge](https://www.elastic.co/guide/en/apm/agent/go/current/opentelemetry.html).

> 

Reviewer: maybe say "please refer to the blog post for further details"?
